components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementServiceImpl.java

```diff
@@ -695,6 +695,34 @@ public int getRolesCount(String tenantDomain) throws IdentityRoleManagementExcep
         return count;
     }
 
+    @Override
+    public int getRolesCount(String searchFilter, String tenantDomain) throws IdentityRoleManagementException {
+
+        List<RoleManagementListener> roleManagementListenerList = RoleManagementServiceComponentHolder.getInstance()
+                .getRoleManagementListenerList();
+        for (RoleManagementListener roleManagementListener : roleManagementListenerList) {
+            if (roleManagementListener.isEnable()) {
+                roleManagementListener.preGetRolesCount(searchFilter, tenantDomain);
+            }
+        }
+        RoleManagementEventPublisherProxy roleManagementEventPublisherProxy =
+                RoleManagementEventPublisherProxy.getInstance();
+        roleManagementEventPublisherProxy.publishPreGetRolesCountWithException(searchFilter, tenantDomain);
+        List<ExpressionNode> expressionNodes = getExpressionNodes(searchFilter);
+        int count = roleDAO.getRolesCount(expressionNodes, tenantDomain);
+        roleManagementEventPublisherProxy.publishPostGetRolesCount(searchFilter, tenantDomain);
+        for (RoleManagementListener roleManagementListener : roleManagementListenerList) {
+            if (roleManagementListener.isEnable()) {
+                roleManagementListener.postGetRolesCount(count, searchFilter, tenantDomain);
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("%s for the filter %s get roles count successfully.", getUser(tenantDomain),
```

Comment:
Let's check this debug log

Reply:
52cdf24c27571a1401a64264c82d95c27a32568a